lib/phoenix/pubsub/pg2_server.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Phoenix.PubSub.PG2Server do
+  use GenServer
+
+  @moduledoc """
+  `Phoenix.PubSub` adapter based on `:pg2`
+
+  See `Phoenix.PubSub.Redis` for details and configuration options.
+  """
+
+  def start_link(opts) do
+    GenServer.start_link __MODULE__, opts, name: Dict.fetch!(opts, :name)
+  end
+
+  def init(opts) do
+    server_name   = Keyword.fetch!(opts, :name)
+    local_name    = Keyword.fetch!(opts, :local_name)
+    pg2_namespace = pg2_namespace(server_name)
+
+    Process.flag(:trap_exit, true)
+
+    :ok = :pg2.create(pg2_namespace)
+    :ok = :pg2.join(pg2_namespace, self)
+
+    {:ok, %{local_name: local_name, namespace: pg2_namespace}}
+  end
+
+  def handle_call({:subscribe, pid, topic, link}, _from, state) do
+    if link, do: Process.link(pid)
+    {:reply, GenServer.call(state.local_name, {:subscribe, pid, topic}), state}
+  end
+
+  def handle_call({:unsubscribe, pid, topic}, _from, state) do
+    {:reply, GenServer.call(state.local_name, {:unsubscribe, pid, topic}), state}
+  end
+
+  def handle_call({:broadcast, from_pid, topic, msg}, _from, state) do
+    case :pg2.get_members(state.namespace) do
+      {:error, {:no_such_group, _}} -> {:stop, :no_global_group, state}
```

Comment:
You can ignore this clause and just do pids when is_list(pids) below. I.e. just let it crash as we assume no_such_group will never be returned. Unless it can be returned on partitions or something, then we maybe should handle it.